tests/server/test_workon.py

```diff
@@ -763,10 +770,100 @@ async def test_workon_deactivate(
+            'WARNING: Some files in the environment are not owned by the root user. To fix this, run `chown -R root:root ` as root.\n'
+            "WARNING: Make sure you exit the current environment by running the 'deactivate' command rather than simply exiting the shell. This ensures the proper permission checks are performed.\n"
+            'WARNING: The inmanta-workon tool should be run as either root or the inmanta user to have write access (to be able to run pip install or inmanta project install).\n'
+            'WARNING: Some files in the environment are not owned by the root user. To fix this, run `chown -R root:root ` as root.\n'
+            "WARNING: Make sure you exit the current environment by running the 'deactivate' command rather than simply exiting the shell. This ensures the proper permission checks are performed."
+        ),
+    )
+
+
+@pytest.mark.slowtest
+async def test_workon_sets_inmanta_config_environment(
+    server: Server,
+    workon_workdir: py.path.local,
+    workon_bash: Bash,
+    workon_environments_dir: py.path.local,
+    compiled_environments: abc.Sequence[data.model.Environment],
+) -> None:
+    """
+    Verify that INMANTA_CONFIG_ENVIRONMENT is correctly reset to its previous value.
+    """
+    outer_env_id: uuid.UUID = uuid.uuid4()
+    inner_env_id: uuid.UUID = compiled_environments[0].id
+    env_dir: py.path.local = workon_environments_dir.join(str(compiled_environments[0].id))
+    # simple deactivate
+    await assert_workon_state(
+        workon_bash,
+        str(inner_env_id),
+        pre_activate=textwrap.dedent(
+            f"""
+            # Set INMANTA_CONFIG_ENVIRONMENT to the outer env's id
+            export INMANTA_CONFIG_ENVIRONMENT={outer_env_id}
+            """.strip(
+                "\n"
+            )
+        ),
+        post_activate=textwrap.dedent(
+            f"""
+            # After activation, verify INMANTA_CONFIG_ENVIRONMENT is set to the inner env's id
+            if [ ! "${{INMANTA_CONFIG_ENVIRONMENT}}" = "{inner_env_id}" ] ; then
+                exit 1
+            fi
+            deactivate
+            # After deactivation, verify INMANTA_CONFIG_ENVIRONMENT has been set back to the outer env's id
+            if [ ! "${{INMANTA_CONFIG_ENVIRONMENT}}" = "{outer_env_id}" ] ; then
+                exit 1
+            fi
+            """.strip(
+                "\n"
+            )
         ),
+        expected_dir=env_dir,
+        invert_python_assert=True,
+        invert_ps1_assert=True,
+        expect_stderr="WARNING: Make sure you exit the current environment by running the 'deactivate' command rather than simply exiting the shell. This ensures the proper permission checks are performed."
+    )
+
+
+
+@pytest.mark.slowtest
+async def test_workon_sets_inmanta_config_environment_empty_outer(
```

Comment:
For this one, perhaps unset the variable in the pre script, just in case it's set on the host running the tests?